File: app/src/main/java/com/simpleshift/scheduler/domain/model/Team.kt
package com.simpleshift.scheduler.domain.model

import androidx.compose.runtime.Immutable

@Immutable
data class Team(
    val id: Int,
    val name: String
) {
    companion object {
        const val TOTAL_TEAMS = 6
        private val CHINESE_NUMERALS = listOf("一", "二", "三", "四", "五", "六")
        val ALL_TEAMS: List<Team> = (1..TOTAL_TEAMS).map { id ->
            Team(id = id, name = "${CHINESE_NUMERALS[id - 1]}值")
        }
    }
}

File: flutter/lib/domain/models/team.dart
/// 班组数据模型 — 对应 Android 版 Team.kt
class Team {
  final int id;

  const Team(this.id);

  static const totalTeams = 6;
  static final List<Team> allTeams =
      List.generate(totalTeams, (i) => Team(i + 1));
}
